packages/flutter/lib/src/widgets/tween_animation_builder.dart

```diff
@@ -117,7 +117,99 @@ class TweenAnimationBuilder<T extends Object?> extends ImplicitlyAnimatedWidget
+  ///
+  /// The animation continuously cycles, restarting from [tween.begin] each time
+  /// it reaches [tween.end]. This is useful for creating animations that need to
+  /// run indefinitely, such as loading spinners, pulsing effects, or continuous
+  /// rotations.
+  ///
+  /// {@template flutter.widgets.TweenAnimationBuilder.repeat.reverse}
+  /// The [reverse] parameter controls the animation direction:
+  ///  * When `false` (default), the animation repeats in one direction, jumping
+  ///    back to the beginning when it completes.
+  ///  * When `true`, the animation reverses direction when it reaches the end,
+  ///    creating a back-and-forth motion.
+  /// {@endtemplate}
+  ///
+  /// {@template flutter.widgets.TweenAnimationBuilder.repeat.paused}
+  /// The [paused] parameter controls whether the animation is running:
+  ///  * When `false` (default), the animation runs continuously.
+  ///  * When `true`, the animation pauses at its current value.
+  ///  * Changing from `true` to `false` resumes the animation from where it paused.
+  /// {@endtemplate}
+  ///
+  /// Unlike the default constructor, [onEnd] is not supported for repeating
+  /// animations since they never end.
+  ///
+  /// {@tool snippet}
+  /// This example shows a spinning square that rotates continuously:
+  ///
+  /// ```dart
+  /// TweenAnimationBuilder<double>.repeat(
+  ///   tween: Tween<double>(begin: 0, end: 1),
+  ///   duration: const Duration(seconds: 2),
+  ///   builder: (BuildContext context, double value, Widget? child) {
+  ///     return Transform.rotate(
+  ///       angle: value * 2 * math.pi,
+  ///       child: Container(
+  ///         width: 100,
+  ///         height: 100,
+  ///         color: Colors.green,
+  ///       ),
+  ///     );
+  ///   },
+  /// )
+  /// ```
+  /// {@end-tool}
+  ///
+  /// {@tool snippet}
+  /// This example shows a pulsing effect using the [reverse] parameter:
+  ///
+  /// ```dart
+  /// TweenAnimationBuilder<double>.repeat(
+  ///   tween: Tween<double>(begin: 0.8, end: 1.2),
+  ///   duration: const Duration(seconds: 1),
+  ///   reverse: true,
+  ///   builder: (BuildContext context, double value, Widget? child) {
+  ///     return Transform.scale(
+  ///       scale: value,
+  ///       child: child,
+  ///     );
+  ///   },
+  ///   child: const Icon(Icons.favorite, color: Colors.red, size: 100),
+  /// )
+  /// ```
+  /// {@end-tool}
+  ///
+  /// See also:
+  ///
+  ///  * [AnimationController.repeat], which this constructor replaces for simple use cases.
+  ///  * [TweenAnimationBuilder], the default constructor for single animations.
+  ///
+  /// ## Ownership
+  ///
+  /// The [TweenAnimationBuilder] takes full ownership of the provided [tween]
+  /// instance and mutates it. Once a [Tween] has been passed to a
+  /// [TweenAnimationBuilder], its properties should not be accessed or changed
+  /// anymore to avoid interference with the [TweenAnimationBuilder].
```

Comment:
This content mirrors but is slightly different from the "Ownership of the Tween" content on TweenAnimationBuilder. I'd consider making a template out of the original content and just use the template here.

https://github.com/flutter/flutter/blob/6cbd98adc517ca7a20c724c4f0c5a5a385b95f74/packages/flutter/lib/src/widgets/tween_animation_builder.dart#L58C1-L67C34

Reply:
the issue of making a template (now) where it is a separate component is that I can't repeat "The [TweenAnimationBuilder] takes full ownership" unless it accepts variable so might make sense to repeat now :( or  just omit or duplicate.